server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessagePreviewListener.java

```diff
@@ -104,12 +109,19 @@ private void preComputePreview(List<MessageId> messageIds, MailboxSession sessio
     }
 
     private Tuple2<MessageId, Preview> computePreview(MessageResult messageResult) throws MailboxException, IOException {
-        Message mimeMessage = messageExtractorHelper.parse(messageResult.getFullContent().getInputStream());
-        MessageContent messageContent = messageExtractorHelper.extractContent(mimeMessage);
-        Preview preview = messageExtractorHelper.mainTextContent(messageContent)
+        Message mimeMessage = parse(messageResult);
+        MessageContent messageContent = messageContentExtractor.extract(mimeMessage);
+        Preview preview = messageContent.mainTextContent(htmlTextExtractor)
             .map(Preview::compute)
-            .orElse(Preview.NO_BODY);
+            .get();
```

Comment:
How can you be sooo  sure this is always here?

If `messageContent.mainTextContent` is not empty it should not return an optional in the first place!